src/mapping/Mapping.hpp

```diff
@@ -46,8 +46,21 @@ class Mapping {
     FULL = 2
   };
 
+  /**
+   * @brief Specifices what kind of integrals to compute for scaled-consistent mapping.
+   * 
+   * Scaled-Consistent mappings conserve integrals, but they must be computed differently 
+   * depending on the nature of the coupling. It can be either SURFACE coupling (e.g. pressure to conserve forces)
+   * or VOLUMETRIC coupling (e.g. density to preserve mass).
```

Comment:
Do you mean by "nature of the coupling" just volume coupling vs surface coupling? In principle, I would expect that the user provides the desired degree of connectivity he wants to apply (and is also able to fetch the information from preCICE).

Internally, one could then distinguish using the space dimension and the size of the connectivity container. If we still want to keep the `enum`, I would hope for a different name, as 'kind' means kind of anything (could also mean explicit, implicit, consistent etc). What about `CouplingGeometry`, `CouplingDimension`, `CouplingEntity`?!

Reply:
The way I see it, the user would provide it in the XML config. It feels safer to me than guessing from the connectivity. Would you want something like `if (mesh->tetrahedra().empty) integrateVolume(...) else integrateSurface()` ? (Assuming 3D here)

I'm not a big fan of your suggestions but I agree that "kind" is not ideal. What about `IntegralType`? Or a boolean `integrateOnBoundary` (defaulting to true) ? That's probably a lot of booleans though since there is already the gradient one, so we'll end up with unreadable calls like Mapping(consistent, 3, false, false)

Reply:
> It feels safer to me than guessing from the connectivity. 

Sounds reasonable

>  What about IntegralType

Maybe `IntegralManifold`? Anyway, feel free to decide on something. It should be as self explanatory as possible.

Reply:
IMO, `IntegralType` should do be understandable here.
In the end "volume vs. surface" only makes a difference for mapping methods that make use of integrals in some sense.
But why is this variable only used for "scaled-consistent mapping", shouldn't it also have an influence on the nearest-projection mapping. Do we at least have safeguard assertions there?

Reply:
NP mapping should be Surface-only, LCI mapping Volume-only, and all the others (NN(G) and RBFs) could be both.
My idea was to have no safeguard internally but safeguards on the API:
- If In precice-config I put volume or surface with NN, _alles gut_
- If I configure NP with volume or LCI with surface, throw an error
- If I configure NP with surface or LCI with volume, either do nothing or warn that configuration wasn't needed.

But in the internals, the scaling is done in the mapping base class, so putting safeguards there would be ugly.

Reply:
But I don't get what you meant by  "shouldn't it also have an influence on the nearest-projection mapping". The point is that it's one of the few mappings where it shouldn't.

Reply:
Mmh, I thought you would put `LCI` in the config and deduce from this that it is a volume coupling.
In the LCI mapping class I would then put an "assertion(type==surface)" to safeguard against future changes / wrong unit tests etc.